Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -7383,7 +7389,12 @@ void HTMLMediaElement::createMediaPlayer() WTF_IGNORES_THREAD_SAFETY_ANALYSIS
     m_player->setPreferredDynamicRangeMode(m_overrideDynamicRangeMode.value_or(preferredDynamicRangeMode(document().view())));
     m_player->setShouldDisableHDR(shouldDisableHDR());
     m_player->setMuted(effectiveMuted());
+#if PLATFORM(IOS_FAMILY)
+    auto page = document().page();
+    m_player->setPageIsVisible(!m_elementIsHidden, page ? page->sceneIdentifier() : ""_s);
```

Comment:
I have no opinion here, but I wonder if it would be better to have the `#if` conditional inside `sceneIdentifier()` rather than the calling site. i.e. `Page::sceneIdentifier` is on all platforms, and returns `""_s` everywhere but IOS_FAMILY.

Reply:
That seems like a better idea, I'll make that change